test/e2e/lifecycle/eg_lifecycle_test.go

```diff
@@ -63,11 +63,30 @@ func TestEGUpgrade(t *testing.T) {
 		t.Fatalf("Failed to create test suite: %v", err)
 	}
 
-	t.Logf("Running %d Upgrade tests", len(tests.UpgradeTests))
-	cSuite.Setup(t, tests.UpgradeTests)
+	reorderLifeCycleTest(&tests.LifecycleTests)
+	t.Logf("Running %d lifecycle tests", len(tests.LifecycleTests))
+	cSuite.Setup(t, tests.LifecycleTests)
 
-	err = cSuite.Run(t, tests.UpgradeTests)
+	err = cSuite.Run(t, tests.LifecycleTests)
 	if err != nil {
 		t.Fatalf("Failed to run tests: %v", err)
 	}
 }
+
+// we reorder lifecycle test, and we want 'Envoy Shutdown' to be the first test to
+// run to avoid being affected.
+func reorderLifeCycleTest(t *[]suite.ConformanceTest) {
```

Comment:
Maybe we can just register all the tests in the suite setup instead of allowing them to register themselves in arbitrary order?